### ISSUE TITLE
Create a separate python package just for REST API functions

### ISSUE THREAD

**Issue by alexott** [delta-io/delta-sharing#169]
Right now the `delta-sharing` package includes dependencies on arrow, pandas, etc. and this makes it's quite heavyweight - installations requires ~50Mb, and that it's not acceptable for use cases, such as, Delta Sharing sensor in the Airflow where we just need to check the latest version of the Delta Share table, and or download data files without reading them.

It would be useful to release more lightweight package that will provide an unified interface to the REST API, for example, `delta-sharing-rest-api`, or something like that

**Comment by linzhou-db** [comment]
I see, this makes sense! Let's discuss the timeline for this.

**Comment by souvik-databricks** [comment]
@linzhou-db 
For this effort of creation of a python SDK around the Delta sharing REST APIs I am happy to help write the package.

**Comment by linzhou-db** [comment]
@souvik-databricks That's great, please go ahead! And I can review the PRs. Let me know if you need anything from me.

**Comment by souvik-databricks** [comment]
hi @alexott 

I see that we have a [rest client for delta sharing](https://github.com/delta-io/delta-sharing/blob/main/python/delta_sharing/rest_client.py) 

So when you say light-weight wrapper for the REST API can you give some more details on the exactly what kind of stuff are we talking about stripping off from the current existing REST client.

**Comment by alexott** [comment]
@souvik-databricks this client is a part of bigger package that has a lot of dependencies that aren't necessary for REST client - arrow, pandas, etc.  It would be useful to have a single package for REST API only that only depends on `requests` package

**Comment by souvik-databricks** [comment]
@alexott 
Got it the ask is basically for a just a wrapper on the REST API which leverages only the `requests` package. Without any of the heavy packages. 

I will start working on this now @linzhou-db.

**Comment by souvik-databricks** [comment]
@alexott The responses of the APIs of this `requests` based package does it have to be modeled through data classes or should I leave the responses in the form of python dictionaries?

@linzhou-db One question I have is should I put the file in a separate package within the python directory or should I keep them in the one file `rest_client_light.py` which will stay besides the current `rest_client.py` file

**Comment by alexott** [comment]
I personally would prefer data classes - primary ask is just to take existing REST API client and release it as a separate package with a minimum of dependencies, similar to [mlflow-skinny](https://pypi.org/project/mlflow-skinny/)

**Comment by linzhou-db** [comment]
yeah, if possible I think we can put the file in a separate package, which is good enough.

**Comment by souvik-databricks** [comment]
Got stuck in a few workstreams. So I might be running a bit late on this work. @linzhou-db

**Comment by chitralverma** [comment]
I think if this is done, we can also work on standardizing the overall API as well. For example, in a lot of if i want to pass `share` it as to be `str` and in other places it has to be `Share`, there are other similar things as well.

**Comment by alexott** [comment]
also, if we're talking about APIs, we may also need to provide async versions of API functions.

**Comment by souvik-databricks** [comment]
Hi @chitralverma I have gotten stuck in couple of other workstreams so I haven't been able to put in my time. 

If you have time you can take this work. 

@linzhou-db Please assign this to @chitralverma
